homeassistant/components/device_tracker/tado.py

```diff
@@ -0,0 +1,109 @@
+"""
+homeassistant.components.device_tracker.tado.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Device tracker platform that supports presence detection.
+The detection is based on geofencing enabled devices used with Tado.
+Tado is a Smart Thermostat device.
+"""
+import logging
+from datetime import timedelta
+from collections import namedtuple
+
+import requests
+import voluptuous as vol
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.components.device_tracker import DOMAIN, PLATFORM_SCHEMA
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string
+})
+
+REQUIREMENTS = []
+
+
+def get_scanner(hass, config):
+    """Validate config and returns a Tado scanner."""
+    info = config[DOMAIN]
+
+    if info.get(CONF_USERNAME) is None or info.get(CONF_PASSWORD) is None:
+        _LOGGER.warning('Cannot find username or password')
+        return None
```

Comment:
Those checks are no longer needed. The validation of the configuration is done by `voluptuous`. Password and username are required by the platform.